bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -491,26 +489,24 @@ public void asyncGetLedgersContainBookies(final Set<BookieSocketAddress> bookies
         bkc.getLedgerManager().asyncProcessLedgers(new Processor<Long>() {
             @Override
             public void process(final Long lid, final AsyncCallback.VoidCallback cb) {
-                bkc.getLedgerManager().readLedgerMetadata(lid, new GenericCallback<Versioned<LedgerMetadata>>() {
-                    @Override
-                    public void operationComplete(int rc, Versioned<LedgerMetadata> metadata) {
-                        if (BKException.Code.NoSuchLedgerExistsException == rc) {
-                            // the ledger was deleted during this iteration.
-                            cb.processResult(BKException.Code.OK, null, null);
-                            return;
-                        } else if (BKException.Code.OK != rc) {
-                            cb.processResult(rc, null, null);
-                            return;
-                        }
-                        Set<BookieSocketAddress> bookiesInLedger = metadata.getValue().getBookiesInThisLedger();
-                        Sets.SetView<BookieSocketAddress> intersection =
+                bkc.getLedgerManager().readLedgerMetadata(lid)
+                    .whenComplete((metadata, exception) -> {
+                            if (exception instanceof BKException.BKNoSuchLedgerExistsException) {
```

Comment:
isn't exception here a `CompletionException`?

Reply:
No, shouldn't be, but it's dependent on how readLedgerMetadata is implemented, so will change to getCode